plugin-commons/src/main/java/org/sonarsource/sonarlint/core/plugin/commons/loading/PluginClassloaderFactory.java

```diff
@@ -90,7 +92,7 @@ private static void exportResources(PluginClassLoaderDef def, ClassloaderBuilder
     for (PluginClassLoaderDef def : defs) {
       var classloader = classloadersByBasePluginKey.get(def.getBasePluginKey());
       if (classloader == null) {
-        throw new IllegalStateException(String.format("Fail to create classloader for plugin [%s]", def.getBasePluginKey()));
+        LOG.error("Fail to create classloader for plugin '{}'", def.getBasePluginKey());
       }
       result.put(def, classloader);
```

Comment:
So now the Map may contain a definition with a `null` classloader. It will fail badly later.

Reply:
Good catch! I should put in the map only if classloader is not null.
Now looking at the code, I am not even sure that classloader could be null at any time here.